### PR TITLE
added solution

### PR DESCRIPTION
**Pull request by VitaliiLeshchenko**

### CODE REVIEW

File: src/main/java/core/basesyntax/Car.java [hunk 1]
```diff
@@ -1,5 +1,16 @@
...
+    private String number;
+    private String bodyType;
+    private String engineType;
+    private Engine engine;
 
+    public Car() {
+    }
+
+    public Car clone() throws CloneNotSupportedException {
+        return (Car)super.clone();
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Car.java:14]: that would work for shallow cloning, but the thing is that we need deep cloning cause we have Engine object with his own fields as an Car's field

Comment (VitaliiLeshchenko) [reply src/main/java/core/basesyntax/Car.java]: changed method clone() in Car class

Comment (VitaliiLeshchenko) [reply src/main/java/core/basesyntax/Car.java]: added another commit with new constructors and rewritten clone method

File: src/main/java/core/basesyntax/Engine.java [hunk 2]
```diff
@@ -1,5 +1,22 @@
 package core.basesyntax;
 
-public class Engine {
+public class Engine implements Cloneable {
+    private int param1;
+    private int param2;
+    private int param3;
+    private int param4;
+    private int param5;
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Engine.java:8]: let's make our Engine t a little bit more realistic)

Comment (VitaliiLeshchenko) [reply src/main/java/core/basesyntax/Engine.java]: changed Engine parameters

**Pull request closed without merging**